tests/integration/test_runner_with_kafka.py

```diff
@@ -123,21 +136,35 @@ def test_write_read_to_kafka(start_service, subtests):
         try:
             runner.run()
         except ray.exceptions.RayActorError:
-            # This is expected because we activated the poison pill
-            pass
-
-        consumer = DatasetConsumer(
-            dataset_name="test_result",
-            node_name="consumer",
-            pipeline_name="integration_test_read",
-            dataset_config={},
-            has_pipeline_prefix=True,
-        )
-        count_messages = consumer.consume_all(end_message="END")
-        assert count_messages[0]["source_pipeline"] == "integration_test_read"
-        assert count_messages[0]["message"] == "TEST PASSED"
+            dataset_name = "test_result"
+            dataset_config = {
+                "type": "aineko.datasets.kafka.Kafka",
+                "location": "localhost:9092",
+            }
+            dataset = AbstractDataset.from_config(dataset_name, dataset_config)
+            consumer_params = ConsumerParams(
+                **{
+                    "dataset_name": dataset_name,
+                    "node_name": "consumer",
+                    "pipeline_name": "integration_test_read",
+                    "prefix": None,
+                    "has_pipeline_prefix": True,
+                    "consumer_config": DEFAULT_KAFKA_CONFIG.get(
+                        "CONSUMER_CONFIG"
+                    ),
+                }
+            )
+            dataset.initialize(
+                create_consumer=True, connection_params=consumer_params
+            )
+            count_messages = dataset.consume_all(end_message="END")
+            print("count_messages are...")
+            print(count_messages)
+            print("...")
+            # assert count_messages[0]["source_pipeline"] == "integration_test_read"
```

Comment:
i added it back in. originally, i wasn't going to implement since we're updating the message structure to remove metadata. but i think that's something that should wait until the dataset validation PR

Reply:
Yes, we shouldn't modify anything that's out of scope for this feature.